kvserver: factor out uninitialized replica creation

pav-kv opened the pull request:
Do not review.

Epic: CRDB-220
Release note: None

pav-kv committed:
tmp

pav-kv committed:
todos

pav-kv committed:
kvserver: factor out uninitialized replica creation

Epic: CRDB-220
Release note: None

pav-kv committed:
kvserver: unindent

Epic: none
Release note: None

pav-kv committed:
fix commenting

pav-kv committed:
don't load uninitialized replica

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94765)
<!-- Reviewable:end -->

pav-kv committed:
kvserver: init CT side transport in newUnloadedReplica

Logically it could be initialized at the load time, but later commits remove
load step for uninitialized replicas, and some tests break if side transport is
not inited right after Replica creation.

Release note: None

pav-kv committed:
kvserver: init SideloadStorage in newUnloadedReaplica

Sideload storage is necessary both for initialized and uninitialized replicas,
but we are removing the load step for uninitialized replicas. Creating the
sideload storage is cheap, so we are moving it to newUnloadedReplica.

Release note: None

pav-kv committed:
kvserver: init range descriptor in newUnloadedReplica

Both callers of newUnloadedReplica initialize its state.Desc before linking it
elsewhere. This change moves this initialization inside newUnloadedReplica.

Release note: none
